x/ecocredit/server/testsuite/suite.go

```diff
@@ -50,14 +54,13 @@ func NewIntegrationTestSuite(fixtureFactory testutil.FixtureFactory, paramSpace
 
 func (s *IntegrationTestSuite) SetupSuite() {
```

Comment:
I think we should consider refactoring this massive integration test into smaller, bite sized unit tests, and then just have a big integration scenario test. 

it's a bit confusing to figure my way around this code. the batch info test is broken and the one thats failing has a hard coded batch denom to search for. i'm not really sure what the best approach to fixing that test would be other than obliterating the existing test structure to fit my needs..

should we create an issue to move these smaller tests into unit tests? @aaronc @ryanchristo @robert-zaremba thoughts?

Reply:
I agree. A lot of these tests are written like unit tests already. I think it makes sense to separate them out and to simply the integration tests in a separate issue. You should be able to trace the batch denom issue without obliterating the existing structure. Would be happy to take a look if you get stuck.